edgecast/internal/ecauth/ids_client.go

```diff
@@ -54,6 +54,21 @@ func (c IDSClient) GetToken(
 		return nil, err
 	}
 
+	if resp.StatusCode == http.StatusBadRequest {
+		errorResponse := &OAuth2ErrorResponse{}
+		err = json.NewDecoder(resp.Body).Decode(&errorResponse)
+		if err != nil {
+			return nil, err
```

Comment:
If the JSON decoder fails, it might be best to convert the Body to a string and just include that raw JSON string as part of the error.

Reply:
Just keep in mind that `resp.Body` can only be read once. So the steps would be:
1. convert resp.Body to a string
2. decode bodyString to OAuth2ErrorResponse
3. If decode fails, throw an error with the raw bodyString